R/run-model.R

```diff
@@ -320,3 +336,35 @@ naomi_info <- function(data, options) {
        "options.yml" = yaml::as.yaml(options),
        "packages.csv" = write_csv_string(naomi_info_packages()))
 }
+
+## Ensures data is of format
+## list(
+##   input = list(
+##     path = "path/to/file",
+##     hash = "file_hash",
+##     filename = "filename"
+##   ),
+##   input2 = list(
+##     ...
+##   ),
+##   ...
+## )
+## From either a list of file paths or returns passed in data unchanged
+format_data_input <- function(data) {
+  if (all(vapply(data, is.character, logical(1)))) {
```

Comment:
Is correct reading of this that it requires all to be either 'path' format or all 'list' format?  If so, should this check be applied to each file separately?

I don't think this is a necessary change right now -- there's not a clear use case for it at present.

Reply:
That's right - all either list or path. Mostly because I agree I can't see a use case for having a mix. Would be fairly simple to add if we end up having a mix sometime in the future.